# Palmer Penguins Data Analysis Project

This notebook contains my analysis of the famous Palmer Penguins dataset.

The data set is available [on GitHub](https://allisonhorst.github.io/palmerpenguins/).



### Imports
*** 

Libraries that will be used to conduct and visualise my data analysis on the dataset;

- [numpy](https://numpy.org/) 

- [pandas](https://pandas.pydata.org/docs/index.html)

- [matplotlib](https://matplotlib.org/stable/)

- [seaborn](https://seaborn.pydata.org/index.html)

In [1]:
# Data frames. 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### Load Data
***
Load the Palmer Pengueins data set from a URL.

In [11]:
# Load the Palmer Penguins data set.
penguins = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv")

In [12]:
# Read the data set.
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    MALE  
1         3800.0  FEMALE  
2         3250.0  FEMALE  
3            NaN     NaN  
4         3450.0  FEMALE  
..           ...     ...  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE  

[344 rows x 7 columns]

We can see here that data set has 344 rows and 7 columns. 

In [13]:
# Looking at the DataFrame information.
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


In [14]:
# Descriptive statistics of the DataFrame.
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [25]:
print(penguins.duplicated().sum()) # Check for duplicates.
penguins.isnull().sum() # Check for missing values.

0


species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

0

In [5]:
# Look at the column "Sex"
df['sex']

0        MALE
1      FEMALE
2      FEMALE
3         NaN
4      FEMALE
        ...  
339       NaN
340    FEMALE
341      MALE
342    FEMALE
343      MALE
Name: sex, Length: 344, dtype: object

In [6]:
# Count the number of penguins of each sex
df['sex'].value_counts()

sex
MALE      168
FEMALE    165
Name: count, dtype: int64

In [7]:
# Describe the data set.
df.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

Basis of MD
# This will be big (headline_1)

We *might* (italic) have a **paragraph** (bold) here. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. 

If you want to second paragraph leave a line in between. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph. Sentences of the paragraph.

## This will be smaller (headline_2)

- bullet
- point
- lists
  - indent
  - bullet
  - points
### This will be smaller again (headline_3)

1. number (if you pur 1 for each line, MD renders and puts them in order)
1. bullet
1. points

#### This will be smaller again again (headline_4)
##### This will be smaller again again again (headline_5)
###### This will be smaller again again again again (headline_6)

## Tables

***
(pipe)
| Species   |Bill Length (mm)|Body Mass (g)|
|-----------|---------------:|------------:|
|Adelie     |            38.8|         3701|
|Chinstrap  |            48.8|         3733|
|Gentoo     |            47.5|         5076|

```python
print("Python is a great language!")

a = a + 1

print(a)
```

## Math

***

f(x) = x2  (wrong)

$f(x) = x^2$ (right)

$\sum_{i=0}^{n-1}$

$\bar{x} = \frac{\sum_{i=0}^{n-1} x_i}{n}$

***
### End